magenta/music/melodies_lib.py

```diff
@@ -857,38 +892,216 @@ def squash_and_encode(self, melody):
 
 
 class OneHotMelodyEncoderDecoder(MelodyEncoderDecoder):
-  """A MelodyEncoderDecoder that produces a one-hot encoding for the input."""
+  """A MelodyEncoderDecoder that produces a one-hot encoding for the input"""
+
+  @property
+  def input_size(self):
+    return self.num_melody_events
+
+  @property
+  def num_classes(self):
+    return self.num_melody_events
+
+  def events_to_input(self, events, position):
+    """Returns the input vector for the given position in the melody.
+
+    Returns a one-hot vector for the given position in the melody mapped to the
+    model's event range. 0 = no event, 1 = note-off event,
+    [2, self.num_classes) = note-on event for that pitch relative to the
+    [self._min_note, self._max_note) range.
+
+    Args:
+      events: A Melody object.
+      position: An integer event position in the melody.
+
+    Returns:
+      An input vector, a list of floats.
+    """
+    input_ = [0.0] * self.input_size
+    input_[self.melody_event_to_index(events[position])] = 1.0
+    return input_
+
+  def events_to_label(self, events, position):
+    """Returns the label for the given position in the melody.
+
+    Returns the zero-based index value for the given position in the melody
+    mapped to the model's event range. 0 = no event, 1 = note-off event,
+    [2, self.num_classes) = note-on event for that pitch relative to the
+    [self._min_note, self._max_note) range.
+
+    Args:
+      events: A Melody object.
+      position: An integer event position in the melody.
+
+    Returns:
+      A label, an integer.
+    """
+    return self.melody_event_to_index(events[position])
+
+  def class_index_to_event(self, class_index, events):
+    """Returns the melody event for the given class index.
+
+    This is the reverse process of the self.events_to_label method.
+
+    Args:
+      class_index: An integer in the range [0, self.num_classes).
+      events: A Melody object. This object is not used in this implementation.
+
+    Returns:
+      A Melody event value.
+    """
+    return self.index_to_melody_event(class_index)
+
+
+class LookbackMelodyEncoderDecoder(MelodyEncoderDecoder):
```

Comment:
Also, advance warning that I'm going to take this and make it non-melody-specific.